### DISCUSSION TITLE
Is there any way to install on another drive instead of "C:/" ?

### DISCUSSION THREAD

**Discussion by hyunki** [LSPosed/MagiskOnWSALocal#178] [Q&A] [answered]
Hi, thank you for the scripts! it is really convenient.

Q1: **Is there anyway that I can move this vhdx or install on D drive?**
My computer has limited space, 500gb ssd on c:/ drive 
D:/ drive has a 4tb instead.
**I want to install wsa on D drive.**
vhdx file seems exist on c:/users/userprofile/appdata/local/packages....

Q2: **Can I install two WSA on my computer?**
I want to run wsa with magisk and another one without magisk

**Comment by Doddddd** [comment]
mklink

**Reply by hyunki** [reply]
> mklink

Symbolic link doesn't help my purpose of reduce space usage in c drive.

**Comment by Howard20181** [comment]
https://learn.microsoft.com/en-us/powershell/module/appx/move-appxpackage

**Reply by hyunki** [reply]
Thank you very much. I'll try this.

**Reply by hyunki** [reply]
"get-appxpackage" is working but "move" doesn't work for me :(

get-appxpackage MicrosoftCorporationII.WindowsSubsystemForAndroid                                                                                                                                                                                                
Name              : MicrosoftCorporationII.WindowsSubsystemForAndroid                                                                     
Publisher         : CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US                                      Architecture      : X64                                                                                                                   
ResourceId        :                                                                                                                       
Version           : 2207.40000.8.0                                                                                                        
PackageFullName   : MicrosoftCorporationII.WindowsSubsystemForAndroid_2207.40000.8.0_x64__8wekyb3d8bbwe                                   InstallLocation   : C:\MagiskWSA\output\WSA-with-magisk-stable-MindTheGapps_2207.40000.8.0_x64_Release-Nightly                            IsFramework       : False                                                                                                                 
PackageFamilyName : MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe                                                       PublisherId       : 8wekyb3d8bbwe                                                                                                         
IsResourcePackage : False                                                                                                                 
IsBundle          : False
IsDevelopmentMode : True
NonRemovable      : False
Dependencies      : {Microsoft.VCLibs.140.00.UWPDesktop_14.0.30704.0_x64__8wekyb3d8bbwe, Microsoft.UI.Xaml.2.6_2.62112.3002.0_x64__8wekyb                     3d8bbwe, Microsoft.VCLibs.140.00_14.0.30704.0_x64__8wekyb3d8bbwe}
IsPartiallyStaged : False
SignatureKind     : None
Status            : Ok

move-appxpackage -package "MicrosoftCorporationII.WindowsSubsystemForAndroid" -volume D:\
move-appxpackage : No element. (Exception HRESULT: 0x80070490)
+ move-appxpackage -package "MicrosoftCorporationII.WindowsSubsystemFor ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Move-AppxPackage], Exception
    + FullyQualifiedErrorId : System.Exception,Microsoft.Windows.Appx.PackageManager.Commands.MoveAppxPackageCommand

move-appxpackage -package "MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe" -volume D:\
move-appxpackage : No element. (Exception  HRESULT: 0x80070490)
+ move-appxpackage -package "MicrosoftCorporationII.WindowsSubsystemFor ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Move-AppxPackage], Exception
    + FullyQualifiedErrorId : System.Exception,Microsoft.Windows.Appx.PackageManager.Commands.MoveAppxPackageCommand

**Reply by Howard20181** [reply]
try `Mount-AppxVolume -Volume D:\` before move

**Reply by Howard20181** [reply]
And try `Add-AppxVolume -Path "D:\WindowsApps"` if it doesn't exist.

**Reply by hyunki** [reply]
> And try `Add-AppxVolume -Path "D:\WindowsApps"` if it doesn't exist.

Add-AppxVolume -Path "D:\WindowsApps" is generated
but still not working. Move command cannot find app package.

move-appxpackage : Deployment failed with HRESULT: 0x80073CF1, Package was not found.
Cannot find MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe in repository so MovePackageOperat
ion. 
NOTE: For additional information, look for [ActivityId] 80c24802-df62-0002-11d2-cf8062dfd801 in the Event Log or use th
e command line Get-AppPackageLog -ActivityID 80c24802-df62-0002-11d2-cf8062dfd801
At line:1 char:1
+ move-appxpackage -package "MicrosoftCorporationII.WindowsSubsystemFor ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (MicrosoftCorpor...d_8wekyb3d8bbwe:String) [Move-AppxPackage], PSInvalid
   OperationException
    + FullyQualifiedErrorId : DeploymentError,Microsoft.Windows.Appx.PackageManager.Commands.MoveAppxPackageCommand

Guess I have to just using this. I consumed almost 8hours to solve this...
Thank you for your kind replies.

**Reply by Howard20181** [reply]
The location of an application deployed in developer mode cannot be changed; its location is the location where it is unpacked.
The location of the data storage cannot be changed.

**Reply by Howard20181** [reply]
Maybe we can mklink to another driver
https://www.windows11tips.com/how-to-customize-windows-subsystem-for-android-file-path-in-windows-11/

**Reply by hyunki** [reply]
> Maybe we can mklink to another driver https://www.windows11tips.com/how-to-customize-windows-subsystem-for-android-file-path-in-windows-11/

```
fsutil behavior set disableencryption 1
fsutil behavior set disablecompression 1
```
```
mklink /j "C:\Users\whitg\AppData\Local\Packages\MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe\LocalCache\userdata.vhdx" "D:\LocalCache"
Cannot create a file when that file already exists.
```

I just deleted whole and re-installed MagiskWSA .
VHDX file cannot be shrink through Hyper-V disk manager nor Diskpart.exe too.
I've worked on this whole day and no solution haha.

**Reply by hyunki** [reply]
![image](https://user-images.githubusercontent.com/25163778/195812253-1c86a9bf-6861-4e28-a783-01ce49e87154.png)

**Reply by Howard20181** [reply]
> > Maybe we can mklink to another driver [windows11tips.com/how-to-customize-windows-subsystem-for-android-file-path-in-windows-11](https://www.windows11tips.com/how-to-customize-windows-subsystem-for-android-file-path-in-windows-11/)
> 
> ```
> fsutil behavior set disableencryption 1
> fsutil behavior set disablecompression 1
> ```
> 
> ```
> mklink /j "C:\Users\whitg\AppData\Local\Packages\MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe\LocalCache\userdata.vhdx" "D:\LocalCache"
> Cannot create a file when that file already exists.
> ```
> 
> I just deleted whole and re-installed MagiskWSA . VHDX file cannot be shrink through Hyper-V disk manager nor Diskpart.exe too. I've worked on this whole day and no solution haha.

You don't need to disable encryption and compression
You should delete `%LOCALAPPDATA%\Packages\MicrosoftCorporationII.WindowsSubsystemForAndroid_8wekyb3d8bbwe\LocalCache` before mklink

**Comment by Howard20181** [comment, marked as answer]
A2: No, you can't. Because MagiskOnWSA share the same APPID with the original one.
